src/platforms/blackpillv2/platform.h

```diff
@@ -29,81 +29,155 @@
+ * The 3rd and 4th arguments to this function are optional.
+ */
+#ifndef ALTERNATIVE_PINOUT              // if ALTERNATIVE_PINOUT is not defined
+#define PINOUT_SWITCH(opt0, ...) (opt0) // select the first argument
+#elif ALTERNATIVE_PINOUT == 1
+#define PINOUT_SWITCH(opt0, opt1, ...) (opt1) // select the second argument
+#elif ALTERNATIVE_PINOUT == 2
+#define PINOUT_SWITCH(opt0, opt1, opt2, ...) (opt2) // select the third argument
+#elif ALTERNATIVE_PINOUT == 3
+#define PINOUT_SWITCH(opt0, opt1, opt2, opt3, ...) (opt3) // select the fourth argument
+#else
+#warning "if-statement entered else, which should not be reachable"
+#endif /* ALTERNATIVE_PINOUT */
+
 /*
  * Important pin mappings for STM32 implementation:
  *   * JTAG/SWD
- *     * PB6: TDI
- *     * PB7: TDO/TRACESWO
- *     * PB8: TCK/SWCLK
- *     * PB9: TMS/SWDIO
- *     * PA6: TRST
- *     * PA5: nRST
+ *     * PB6 or PB5: TDI
+ *     * PB7 or PB6: TDO/TRACESWO
+ *     * PB8 or PB7: TCK/SWCLK
+ *     * PB9 or PB8: TMS/SWDIO
+ *     * PA6 or PB3: TRST
+ *     * PA5 or PB4: nRST
  *   * USB USART
  *     * PA2: USART TX
  *     * PA3: USART RX
  *   * +3V3
- *     * PA1: power pin
+ *     * PA1 or PB9: power pin
  *   * Force DFU mode button:
  *     * PA0: user button KEY
  */
 
 /* Hardware definitions... */
+// Build the code using `make PROBE_HOST=blackpillv2 ALTERNATIVE_PINOUT=1` to select the second pinout.
 #define TDI_PORT GPIOB
-#define TDI_PIN  GPIO6
+#define TDI_PIN  PINOUT_SWITCH(GPIO6, GPIO5)
 
 #define TDO_PORT GPIOB
-#define TDO_PIN  GPIO7
+#define TDO_PIN  PINOUT_SWITCH(GPIO7, GPIO6)
 
 #define TCK_PORT   GPIOB
-#define TCK_PIN    GPIO8
+#define TCK_PIN    PINOUT_SWITCH(GPIO8, GPIO7)
 #define SWCLK_PORT TCK_PORT
 #define SWCLK_PIN  TCK_PIN
 
 #define TMS_PORT   GPIOB
-#define TMS_PIN    GPIO9
+#define TMS_PIN    PINOUT_SWITCH(GPIO9, GPIO8)
 #define SWDIO_PORT TMS_PORT
 #define SWDIO_PIN  TMS_PIN
 
-#define TRST_PORT GPIOA
-#define TRST_PIN  GPIO6
+#define TRST_PORT PINOUT_SWITCH(GPIOA, GPIOB)
+#define TRST_PIN  PINOUT_SWITCH(GPIO6, GPIO3)
 
-#define NRST_PORT GPIOA
-#define NRST_PIN  GPIO5
+#define NRST_PORT PINOUT_SWITCH(GPIOA, GPIOB)
+#define NRST_PIN  PINOUT_SWITCH(GPIO5, GPIO4)
 
-#define PWR_BR_PORT GPIOA
-#define PWR_BR_PIN  GPIO1
+#define PWR_BR_PORT PINOUT_SWITCH(GPIOA, GPIOB)
+#define PWR_BR_PIN  PINOUT_SWITCH(GPIO1, GPIO9)
 
 #define LED_PORT       GPIOC
 #define LED_IDLE_RUN   GPIO13
 #define LED_ERROR      GPIO14
 #define LED_BOOTLOADER GPIO15
 
 #define LED_PORT_UART GPIOA
-#define LED_UART      GPIO4
+#define LED_UART      PINOUT_SWITCH(GPIO4, GPIO1)
 
-/* for USART2, DMA1 is selected from https://www.st.com/resource/en/reference_manual/dm00119316-stm32f411xc-e-advanced-arm-based-32-bit-mcus-stmicroelectronics.pdf, page 170, table 27.
- * This table defines USART2_TX as stream 6, channel 4, and USART2_RX as stream 5, channel 4.
+/* USART2 with PA2 and PA3 is selected as USBUSART. Alternatively USART1 with PB6 and PB7 can be used.
  */
```

Comment:
This should be moved onto the line above

Reply:
Implemented 👍